cpupower/cpupower.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Pavithra D P <pavithra@linux.vnet.ibm.com>
+
+import random
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+
+
+class cpupower(Test):
+
+    """
+    Testing cpupower command
+    """
+
+    def test(self):
+        self.nfail = 0
+        self.i = 0
+        self.log.info("Get the initial values from the system")
+        (initial_freq, initial_governor) = self.get_initial_freq()
+        governors = self.get_list_governors()
+        (min, max, cur) = self.get_freeq()
+        for gov in governors:
+            if (gov == "performance"):
+                self.log.info("Check for performance governor")
+                self.check_performance_governor(gov, max)
+            if (gov == "powersave"):
+                self.log.info("Check for powersave governor")
+                self.check_powersave_governor(gov, min)
+            if (gov == "userspace"):
+                self.log.info("Check for userspace governor")
+                self.check_userspace_governor(gov)
+            if (gov == "ondemand" or "conservative" or "schedutils"):
```

Comment:
it would be good to have a set_governor() and check_governor() as generic functions so that lot of if else can be avoided